w3af/core/data/parsers/doc/url.py

```diff
@@ -561,7 +565,36 @@ def is_valid_domain(self):
         """
         :return: Returns a boolean that indicates if self.netloc domain is valid
         """
-        return self.IS_VALID_DOMAIN_RE.match(self.netloc) is not None
+        parsed_url = urlparse(self.netloc)
+    
+        # scheme, netloc must be defined
+        if not all([parsed_url.scheme, parsed_url.netloc]):
+            return False
+        # params, query, fragment must not be defined
+        if any([parsed_url.params, parsed_url.query, parsed_url.fragment]):
```

Comment:
Is this check really necessary? Doesn't `self.netloc` contain only scheme+domain+port because we already parsed it when creating the `URL` instance?

Reply:
Yes, I realized this later.
I'm sorry - I'm reworking the code.